cmd/mt-gateway/main.go

```diff
@@ -0,0 +1,92 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"github.com/grafana/metrictank/logger"
+	log "github.com/sirupsen/logrus"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+	"runtime"
+)
+
+var (
+	version       = "(none)"
+	showVersion   = flag.Bool("version", false, "print version string")
+	metrictankUrl = flag.String("metrictank-url", "http://localhost:6060", "metrictank address")
+	graphiteURL   = flag.String("graphite-url", "http://localhost:8080", "graphite-api address")
+	importerURL   = flag.String("importer-url", "", "mt-whisper-importer-writer address")
+	addr          = flag.String("addr", ":80", "http service address")
+	urls          = Urls{}
+)
+
+type Urls struct {
+	metrictank   *url.URL
+	graphite     *url.URL
+	bulkImporter *url.URL
+}
+
+func init() {
+	formatter := &logger.TextFormatter{}
+	formatter.TimestampFormat = "2006-01-02 15:04:05.000"
+	log.SetFormatter(formatter)
+	log.SetLevel(log.InfoLevel)
+}
+
+func main() {
+	flag.Usage = func() {
+		fmt.Println("mt-gateway")
+		fmt.Println("Provides an HTTP gateway for interacting with metrictank, including metrics ingestion")
+		fmt.Println()
+		fmt.Println("Usage:")
+		fmt.Println()
+		fmt.Println("	mt-gateway [flags]")
+		fmt.Println()
+		fmt.Println("Flags:")
+		flag.PrintDefaults()
+	}
+	flag.Parse()
+
+	if *showVersion {
+		fmt.Printf("mt-gateway (version: %s - runtime: %s)\n", version, runtime.Version())
+		return
+	}
+	var err error
+
+	urls.metrictank, err = url.Parse(*metrictankUrl)
+	if err != nil {
+		log.Fatal(err)
+	}
+	urls.graphite, err = url.Parse(*graphiteURL)
+	if err != nil {
+		log.Fatal(err)
+	}
+	urls.bulkImporter, err = url.Parse(*importerURL)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	metrictankProxy := httputil.NewSingleHostReverseProxy(urls.metrictank)
+	http.HandleFunc("/metrics/index.json", metrictankProxy.ServeHTTP)
+	//TODO drop `/graphite` prefix... Or just drop this entry, do we really need the alias?
+	http.HandleFunc("/graphite/metrics/index.json", metrictankProxy.ServeHTTP)
```

Comment:
Thoughts? can we drop this alias, or is it required for compatibility?

Reply:
I'd leave it for now, because in the "hosted metrics" page which customers see in their account we tell them that the graphite-compatible query api is under the `/graphite` prefix.

Reply:
This shouldn't impact hosted metrics users.

My understanding is that the tsdb-gw will continue to be used for hosted offerings.

This utility is only intended to be used by customers who want to self host.

Reply:
ah, ic. i guess that gives us more freedom to modify (improve) the API then

Reply:
This wont work.  NewSingleHostReverseProxy will pass the origin url path to metrictank.  This will include the '/graphite' prefix.  Metrictank will then just respond with a 404.

Reply:
Yep, that was a placeholder, fixed in b5217edc7e5ba1bcd0574a21e71b792ef9f813a5...

That brings me back to the original question though @woodsaj , do we want to maintain the alias in this version (both `/metrics/index.json` and `/graphite/metrics/index.json` do the same thing)?  Just checking if we want to take the opportunity to clean things up.

Reply:
we should get rid of the graphite prefix.
We should just define routes for everything that is not destined for graphite, then route "/" to graphite.

Reply:
>we should get rid of the graphite prefix.
We should just define routes for everything that is not destined for graphite, then route "/" to graphite.

done in  9c44e7755af7de3244334a020b47b37cbede1cc7